### ISSUE TITLE
Motor Reset when I use attach function.

### ISSUE THREAD

**Issue by gargakshit** [netlabtoolkit/VarSpeedServo#14]
# My servo comes to 0 when I use attach function.
## Here is the code I used:

#include <VarSpeedServo.h>

VarSpeedServo test;

void setup() {
  // put your setup code here, to run onc
  test.attach(2);
}

void loop() {
  // put your main code here, to run repeatedly:
}

**Comment by MatanRubin** [comment]
I'm seeing the same issue - servo resets position when calling `attach()`, instead of just staying where it is. Except for this issue this library is great - exactly what I was looking for - thanks!

**Comment by jackdewhurst** [comment]
Same here, unfortunately this makes the library unusable for my project :-(.

**Comment by gargakshit** [comment]
True. I, because of this library, lost a gold medal in robotics competition.

On Jul 5, 2017 8:29 PM, "Jack Dewhurst" <notifications@github.com> wrote:

Same here, unfortunately this makes the library unusable for my project :-(.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/netlabtoolkit/VarSpeedServo/issues/14#issuecomment-313129141>,
or mute the thread

**Comment by gargakshit** [comment]
# ISSUE

### This problem also occurs with the standard Servo library, also this library is an extension of the Standard Servo library, so the developer might not fix it...

### I got a fix. Get the approximate location where the servo is and write that value after it in the setup() function after calling attach() function. It worked for me...

# I am closing the ISSUE here, comments are still open. People are requested to keep the thread alive...

**Comment by gargakshit** [comment]
Test

**Comment by pvanallen** [comment]
Hi @gargakshit, thanks for clarifying, and offering a work-around. I'm the owner of the library. I will admit that I have not had time recently to work on this repo. I'm am looking for someone to help resolve some of the other open issues and pull requests.

**Comment by gargakshit** [comment]
@pvanallen Sir, should I re-open the issue???

**Comment by jackdewhurst** [comment]
@gargakshit this issue does not occur for me using the standard Servo library.

When I use **write()** directly after **attach()** it still moves to 0 before moving to the desired value, even when wait = false. This doesn't happen using the Servo library.

I've remedied it by exposing **DEFAULT_PULSE_WIDTH** from the library, reading the servo position then setting it before calling **attach()**.

**Comment by gargakshit** [comment]
@jackdewhurst Hey, thanks for the update. This solved the issue. 


### Thanks Again 

_Sent from my YU5010 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by pvanallen** [comment]
@jackdewhurst, @gargakshit  I've reopened this issue - Jack, could you submit a pull request and I'll integrate it? Or just post the changes. Thanks!

**Comment by gargakshit** [comment]
**What's the status???**

**Comment by pvanallen** [comment]
@gargakshit Could you post the code you used to resolve this?

**Comment by cape** [comment]
Hi, I am having the same issue. 
when attach a servo, the servo moves to 0. I need to avoid this, thanks

**Comment by sapics** [comment]
I create the PR to fix this issue.
I have not tested the PR yet, because I don't have servo motor.
It would be helpful, if you could test it.
